### PR TITLE
# 157507769 Setup pre commit hooks

### PR DESCRIPTION
**Pull request by bahdcoder**
#### What does this PR do?
Set up Pre-commit hooks for the project

#### Description of Task to be completed?
- install lint-staged package
- install husky package for setting up hooks
- add lint-staged configurations in package.json file

#### How should this be manually tested?

- clone project
- install dependencies
- Edit the `src/pages/AuthPage/index.tsx` file to have a linting error (for example remove a semicolon)
- Edit the `src/pages/AuthPage/AuthPage.scss` file to have scss linting errors (for example define a style with a color name and not hex value)
- Edit other files randomly.
- Stage these two files only.
- Try to commit.

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#157507769](https://www.pivotaltracker.com/n/projects/2170023/stories/157507769)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,22 +42,29 @@
...
     "webpack-dev-server": "^3.1.4",
     "webpack-merge": "^4.1.2"
+  },
+  "lint-staged": {
+    "*.ts": "npm run lint:ts",
+    "*.tsx": "npm run lint:ts",
+    "*.scss": "npm run lint:scss"
   }
-}
+}
```
Comment (missating) [inline package.json]: @bahdcoder Please fix the end of file mark

**Comment by andela-kakpobome** [comment]
Pretty neat approach to this too, 💯

File: package.json (hunk 1 above)
Comment (bahdcoder) [reply package.json]: Done @missating .

File: package.json [hunk 2]
```diff
@@ -41,22 +42,29 @@
...
     "typescript": "^2.8.3",
     "uglifyjs-webpack-plugin": "^1.2.5",
     "webpack-cli": "^2.1.3",
     "webpack-dev-server": "^3.1.4",
     "webpack-merge": "^4.1.2"
+  },
+  "lint-staged": {
+    "*.ts": "npm run lint:ts",
+    "*.tsx": "npm run lint:ts",
+    "*.scss": "npm run lint:scss"
```
[the remark this answers is not in the document]
Comment (bahdcoder) [reply package.json]: Done @andela-kakpobome .

**Pull request merged**